xmake/modules/private/action/run/make_runenvs.lua

```diff
@@ -35,24 +35,30 @@ function _make_runpath_on_windows(target)
         end
     end
 
-    for _, linkdir in ipairs(target:get("linkdirs")) do
-        insert(linkdir)
-    end
-    for _, opt in ipairs(target:orderopts()) do
-        for _, linkdir in ipairs(opt:get("linkdirs")) do
+    local function insert_target(target)
+        for _, linkdir in ipairs(target:get("linkdirs")) do
             insert(linkdir)
         end
-    end
-    for _, pkg in ipairs(target:orderpkgs()) do
-        for _, linkdir in ipairs(pkg:get("linkdirs")) do
-            insert(linkdir)
+        for _, opt in ipairs(target:orderopts()) do
+            for _, linkdir in ipairs(opt:get("linkdirs")) do
+                insert(linkdir)
+            end
         end
-    end
-    for _, dep in ipairs(target:orderdeps()) do
-        if dep:kind() == "shared" then
-            insert(dep:targetdir())
+        for _, pkg in ipairs(target:orderpkgs()) do
+            for _, linkdir in ipairs(pkg:get("linkdirs")) do
+                insert(linkdir)
+            end
+        end
+        for _, dep in ipairs(target:orderdeps()) do
+            if dep:kind() == "shared" then
+                insert(dep:targetdir())
+            end
+            insert_target(dep)
```

Comment:
If two deps have the same subdeps, this will cause them to be added repeatedly. We should do some target de-duplication to avoid adding the same dep target repeatedly

Reply:
Didn't think of that, this should fix it.